nupic/research/frameworks/vernon/mixins/hidden_activations.py

```diff
@@ -0,0 +1,163 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import abc
+from copy import deepcopy
+from pprint import pformat
+
+import torch
+
+from nupic.research.frameworks.dendrites import plot_hidden_activations_by_unit
+from nupic.research.frameworks.pytorch.hooks import (
+    ModelHookManager,
+    TrackHiddenActivationsHook,
+)
+from nupic.research.frameworks.pytorch.model_utils import filter_modules
+
+__all__ = [
+    "PlotHiddenActivations",
+]
+
+
+class PlotHiddenActivations(metaclass=abc.ABCMeta):
```

Comment:
I'm glad this pattern of hook management and plotting is useful. There's definitely some repeated logic here. We don't need to worry about this much now. If anything, it would be useful however to include a test for this mixin just in case if we try and refactor the code to reduce repeated logic.

Reply:
See `dendrite_metric_mixins_test.py` for an example test. You can even just add to that file if you'd like.

Reply:
I'll add some test cases before merging.

Reply:
Test cases have been added.